src/neo-vm/ExecutionEngine.cs

```diff
@@ -42,6 +85,147 @@ public void AddBreakPoint(byte[] script_hash, uint position)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxItemSize(int length) => length <= MaxItemSize;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxInvocationStack
+        /// </summary>
+        /// <param name="stack">Stack</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckMaxInvocationStack() => InvocationStack.Count < MaxInvocationStackSize;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="value">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigInteger(BigInteger value) => value.ToByteArray().Length <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the BigInteger is allowed for numeric operations
+        /// </summary>
+        /// <param name="byteLength">Value</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckBigIntegerByteLength(int byteLength) => byteLength <= MaxSizeForBigInteger;
+
+        /// <summary>
+        /// Check if the number is allowed from SHL and SHR
+        /// </summary>
+        /// <param name="shift">Shift</param>
+        /// <returns>Return True if are allowed, otherwise False</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckShift(int shift) => shift <= Max_SHL_SHR && shift >= Min_SHL_SHR;
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxStackSize
+        /// </summary>
+        /// <param name="isStackitemCountStrict">Is stack count strict?</param>
+        /// <param name="count">Stack item count</param>
+        /// <returns></returns>
+        public bool CheckStackSize(bool isStackitemCountStrict, int count = 1)
+        {
+            this.is_stackitem_count_strict = isStackitemCountStrict;
+
+            return CheckStackSize(count);
+        }
+
+        /// <summary>
+        /// Check if the is possible to overflow the MaxStackSize
+        /// </summary>
+        /// <param name="stackitem_count">Stack item count</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool CheckStackSize(int count = 1)
+        {
+            stackitem_count += count;
+
+            if (stackitem_count <= MaxStackSize) return true;
+            if (is_stackitem_count_strict) return false;
+            stackitem_count = GetItemCount(InvocationStack.SelectMany(p => p.EvaluationStack.Concat(p.AltStack)));
+            if (stackitem_count > MaxStackSize) return false;
+            is_stackitem_count_strict = true;
```

Comment:
@erikzhang  there should be `false`? because is already "strict counted"

Reply:
After `GetItemCount()` is called, `is_stackitem_count_strict` should be set to `true` because the stack items are strict counted.